presto-main/src/main/java/com/facebook/presto/operator/SimplePagesHashStrategy.java

```diff
@@ -72,10 +76,19 @@ public int getChannelCount()
     @Override
     public long getSizeInBytes()
     {
-        return INSTANCE_SIZE + channels.stream()
-                .flatMap(List::stream)
-                .mapToLong(Block::getRetainedSizeInBytes)
-                .sum();
+        // channels is a List created with ImmutableList#copyOf.
+        // Each element of the List is an ObjectArrayList<Block>.
+        // Hence, the memory used by channels is estimated as
+        //      <header-size-of-ImmutableList> +
+        //      <instance-size-of-ObjectArrayList> * <length-of-outer-list> +
```

Comment:
This still has some % error. Do we need to include `ObjectArrayList#a` size too?
(imagine: channels is single element, channels[0] is very long list of empty blocks. we would calculate O(1) size, ignoring channels[0].a size)